File: src/test/java/compiler/block/operators/AddBlock.java
package compiler.block.operators;

public class AddBlock extends Block {

    private String type = "add";
    private String value;
    private String name;

    public AddBlock(Block superBlock, String name, String value) {
        super(superBlock, false, false);
        this.name = name;
        this.value = value;
        setId(SymbolTable.getInstance().getValue(Utils.getMethod(this), name).getId());
    }

    @Override
    public void init() {

    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public String getValue() {
        return value;
    }

    public void setValue(String value) {
        this.value = value;
    }

    public String getType() {
        return type;
    }

    public void setType(String type) {
        this.type = type;
    }

    @Override
    public String getOpeningCode() {
        return "";
    }

    @Override
    public String getBodyCode() {
        return "mv.visitLdcInsn(" + value + ");\n" +
                "mv.visitVarInsn(ILOAD," + getId() + ");\n" +
                "mv.visitInsn(IADD);\n" +
                "mv.visitVarInsn(ISTORE," + getId() + ");\n";
    }

    @Override
    public String getClosingCode() {
        return "";
    }
}

File: src/main/scala/test_classes/block/operators/AddBlock.scala
package test_classes.block.operators

import test_classes.Utils
import test_classes.block.Block
import test_classes.symbol_table.SymbolTable

/**
  * Represents adding a value to a primitive
  *
  * @param superBlockInit
  * @param name
  * @param valueInit
  */
class AddBlock(var superBlockInit: Block, var name: String, var value: String) extends Block(superBlockInit, false, false) {

  def init() {
    id = (new Integer(SymbolTable.getInstance.getValue(Utils.getMethod(this), name).getId))
  }

  def getName: String = name

  def getType: String = "add"

  def getOpeningCode: String = {
    return "mv.visitIincInsn(" + id + ", " + this.getValue + ");"
  }

  def getValue: String = value

  def getClosingCode: String = {
    return ""
  }

  override def toString: String = "add: " + name

}